azd-templates/infra/scripts/setup-backend.sh

```
@@ -0,0 +1,289 @@
+
+    # Help flag
+    if [[ "${1:-}" == "-h" ]] || [[ "${1:-}" == "--help" ]]; then
+        usage
+    fi
+
+    # Generate resource names
+    # Storage account names must be 3-24 chars, lowercase alphanumeric only
+    local base_name
+    base_name=$(echo "$env_name" | tr '[:upper:]' '[:lower:]' | tr -cd 'a-z0-9')
+    local rg_name="rg-tfstate-${base_name}"
+    local sa_name="tfstate${base_name}$(date +%s | tail -c 5)"  # Add unique suffix
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)